src/replica/ConfigTestApp.cc

```diff
@@ -881,131 +875,92 @@ bool ConfigTestApp::_testTables() {
-                            (databaseInfo.latitudeColName.at(table12) == latitudeColName12) &&
-                            (databaseInfo.longitudeColName.at(table12) == longitudeColName12) &&
-                            (databaseInfo.columns.at(table12).size() == coldefs12.size());
+        bool passed = (databaseInfo.tables().size() == 3) && databaseInfo.tableExists(table12.name) &&
+                      (databaseInfo.findTable(table12.name) == table12);
         cout << (passed ? PASSED_STR : FAILED_STR) << " EXACTLY 3 TABLES SHOULD EXIST NOW"
              << "\n";
         dumpDatabasesAsTable(indent, "");
         success = success && passed;
     }
 
     // Adding the second dependent table connected to the second director.
-    string const table22 = "dependent-2-of-2";
-    bool const isPartitioned22 = true;
-    bool isDirector22 = false;
-    string const directorTable22 = "director-2";
-    string const directorTableKey22 = "director_id_key";
-    string const latitudeColName22 = "decl";
-    string const longitudeColName22 = "ra";
-    list<SqlColDef> coldefs22;
-    coldefs22.emplace_back(directorTableKey22, "INT UNSIGNED");
-    coldefs22.emplace_back(latitudeColName22, "DOUBLE");
-    coldefs22.emplace_back(longitudeColName22, "DOUBLE");
-
-    success = success && addTable(database, table22, isPartitioned22, isDirector22, directorTable22,
-                                  directorTableKey22, latitudeColName22, longitudeColName22, coldefs22);
-    {
-        DatabaseInfo const databaseInfo = config()->databaseInfo(database);
-        auto const tables = databaseInfo.tables();
-        bool const passed = (tables.size() == 4) &&
-                            (find(tables.cbegin(), tables.cend(), table22) != tables.cend()) &&
-                            (databaseInfo.isPartitioned(table22) == isPartitioned22) &&
-                            (databaseInfo.isDirector(table22) == isDirector22) &&
-                            (databaseInfo.directorTable.at(table22) == directorTable22) &&
-                            (databaseInfo.directorTableKey.at(table22) == directorTableKey22) &&
-                            (databaseInfo.latitudeColName.at(table22) == latitudeColName22) &&
-                            (databaseInfo.longitudeColName.at(table22) == longitudeColName22) &&
-                            (databaseInfo.columns.at(table22).size() == coldefs22.size());
+    TableInfo table22;
+    table22.name = "dependent-2-of-2";
+    table22.isPartitioned = true;
+    table22.directorTable = "director-2";
+    table22.directorTableKey = "director_id_key";
+    table22.latitudeColName = "decl";
+    table22.longitudeColName = "ra";
+    table22.columns.emplace_back(table22.directorTableKey, "INT UNSIGNED");
+    table22.columns.emplace_back(table22.latitudeColName, "DOUBLE");
+    table22.columns.emplace_back(table22.longitudeColName, "DOUBLE");
+
+    success = success && addTable(table22);
+    {
```

Comment:
table1of2, table2of2, or table1st, tabl2nd would probably be slightly better names. Just looking at the names I expect 20 more tables.

Reply:
Good point! I will fix that.